addons/l10n_nl_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,150 @@
+        if self._is_nlcius(filename, tree):
+            return self._decode_en_16931(tree, self.env['account.move'])
+        return super()._create_invoice_from_xml_tree(filename, tree)
+
+    def _update_invoice_from_xml_tree(self, filename, tree, invoice):
+        self.ensure_one()
+        if self._is_nlcius(filename, tree):
+            return self._decode_en_16931(tree, invoice)
+        return super()._update_invoice_from_xml_tree(filename, tree, invoice)
+
+    ####################################################
+    # Account_edi_ubl override
+    ####################################################
+    def _get_ubl_partner_values(self, partner):
+        # OVERRIDE
+        values = super()._get_ubl_partner_values(partner)
+        endpoint = partner.l10n_nl_oin or partner.l10n_nl_kvk
+        if partner.country_code == 'NL' and endpoint:
+            scheme = '0190' if partner.l10n_nl_oin else '0106'
+            values['en_16931_endpoint'] = endpoint
+            values['en_16931_endpoint_scheme'] = scheme
+            values['legal_entity'] = endpoint
+            values['legal_entity_scheme'] = scheme
+            values['partner_identification'] = endpoint
+
+        return values
+
+    def _get_ubl_values(self, invoice):
+        values = super()._get_ubl_values(invoice)
+        if self.code != 'nlcius_1':
+            return values
+
+        values['customization_id'] = 'urn:cen.eu:en16931:2017#compliant#urn:fdc:nen.nl:nlcius:v1.0'
+        values['payment_means_code'] = 30
+        values['partner_template'] = 'l10n_nl_edi.export_en_16931_invoice_partner'
+        return values
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_nlcius(self, filename, tree):
+        if self.code != 'nlcius_1':
+            return False
+        customization_id = self._find_value("//*[local-name()='CustomizationID']", tree)
+        return 'nlcius' in customization_id
+
+    def _decode_en_16931(self, tree, invoice):
+        res = super()._decode_en_16931(tree, invoice)
+        if self.code != 'nlcius_1':
+            return res
+
+        if not res.partner_id:
+            namespaces = self._get_ubl_namespaces(tree)
+            endpoint = tree.xpath('//cac:AccountingSupplierParty/cac:Party//cbc:EndpointID', namespaces=namespaces)
+            if endpoint:
+                endpoint = endpoint[0]
+                scheme = endpoint.attrib['schemeID']
+                domains = []
+                if scheme == '0106' and endpoint.text:
+                    domains.append([('l10n_nl_kvk', '=', endpoint.text)])
+                elif scheme == '0190' and endpoint.text:
+                    domains.append([('l10n_nl_oin', '=', endpoint.text)])
+                if domains:
+                    partner = self.env['res.partner'].search(expression.OR(domains), limit=1)
+                    if partner:
+                        res.partner_id = partner
```

Comment:
:boom: ... You can't do that outside the `Form` emulator. This is making your invoice unconsistent :/

Reply:
Added a possibility to have additional domain instead for the partner.